src/metatrain/utils/additive/composition.py

```diff
@@ -91,8 +93,12 @@ def _get_dataloader(
         precision is enforced.
         """
         # Create the collate function
-        targets_keys = list(self.dataset_info.targets.keys())
-        collate_fn = CollateFn(target_keys=targets_keys)
+        collate_fn = CollateFn(
+            target_keys=list(self.dataset_info.targets.keys()),
+            callables=[
+                get_system_with_neighbor_lists_transform(requested_neighbor_lists)
```

Comment:
The composition model should not need a neighbor list, right?

Reply:
I think the idea is to remove the composition inside the dataloader, to hide the corresponding latency